python/hsfs/engine/spark.py

```diff
@@ -532,8 +532,12 @@ def get_training_data(
 
     def split_labels(self, df, labels, dataframe_type):
         if labels:
-            labels_df = df.select(*labels)
-            df_new = df.drop(*labels)
+            if isinstance(df, pd.DataFrame):
```

Comment:
Why is this needed? Was it not working before? Or do you need this for polars instead of pandas?

Reply:
"This behaviour was changed during the implementation of Polars. And dataframe were directly read as Pandas/Polars and Spark dataframe based on the dataframe_type."
It seems like it could break a lot of things. You mean that now you do things that were being done with spark datafra me directly using polars or pandas dataframe?